src/main/java/org/jenkinsci/plugins/credentialsbinding/Binding.java

```diff
@@ -86,25 +70,40 @@ public String getCredentialsId() {
 
     }
 
+    @Deprecated
```

Comment:
Signature change in preparation for [JENKINS-26051](https://issues.jenkins-ci.org/browse/JENKINS-26051), while I am refactoring the API anyway.